content/docs/rfcs/46/README.md

```diff
@@ -0,0 +1,185 @@
+
+Messages are sent over Tor via [SOCKS5](https://www.rfc-editor.org/rfc/rfc1928).
+Tor push uses a dedicated libp2p context to prevent information leakage.
+To significantly increase resilience and mitigate circuit failures,
+Tor push establishes several connections, each to a different randomly selected gossipsub node.
+
+# Specification
+
+This section specifies the format of Tor push messages, as well as how Tor push messages are received and sent, respectively.
+
+## Wire Format
+
+The wire format of a Tor push message corresponds verbatim to a typical [libp2p pubsub message](https://github.com/libp2p/specs/tree/master/pubsub#the-message).
+
+```
+message Message {
+  optional string from = 1;
+  optional bytes data = 2;
+  optional bytes seqno = 3;
+  required string topic = 4;
+  optional bytes signature = 5;
+  optional bytes key = 6;
+}
+```
+
+## Receiving Tor Push Messages
+
+Any node supporting a protocol with ID `/meshsub/1.1.0` (e.g. gossipsub), can receive Tor push messages.
+Receiving nodes are oblivious to Tor push and will process incoming messages according to the respective `meshsub/1.1.0` specification.
+
+## Sending Tor Push Messages
+
+In the following, we refer to nodes sending Tor push messages as Tp-nodes (Tor push nodes).
+
+Tp-nodes MUST setup a separate libp2p context, i.e. [libp2p switch](https://docs.libp2p.io/concepts/multiplex/switch/),
+which MUST NOT be used for any purpose other than Tor push.
+We refer to this context as Tp-context.
+The Tp-context MUST NOT share any data, e.g. peer lists, with the default context.
+
+Tp-peers are peers a Tp-node plans to send Tp-messages to.
+Tp-peers MUST support `/meshsub/1.1.0`.
+For retrieving Tp-peers, Tp-nodes SHOULD use an ambient peer discovery method that retrieves a random peer sample (from the set of all peers), e.g. [33/WAKU2-DISCV5](/spec/33/).
+
+Tp-nodes MUST establish a connection as described in sub-section [Tor Push Connection Establishment](#connection-establishment) to at least one Tp-peer.
+To significantly increase resilience, Tp-nodes SHOULD establish Tp-connections to `D` peers,
+where `D` is the [desired gossipsub out-degree](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.0.md#parameters),
+with a default value of `8`.
+
+Each Tp-message MUST be sent via the Tp-context over at least one Tp-connection.
+To increase resilience, Tp-messages SHOULD be sent via the Tp-context over all available Tp-connections.
+
+Control messages of any kind, e.g. gossipsub graft, MUST NOT be sent via Tor push.
+
+### Connection Establishment
+
+Tp-nodes establish a `/meshsub/1.1.0` connection to tp-peers via [SOCKS5](https://www.rfc-editor.org/rfc/rfc1928) over [Tor](https://www.torproject.org/).
+
+Establishing connections, which in turn establishes the respective Tor circuits, can be done ahead of time.
+
+### Epochs
+
+Tor push introduces epochs.
+The default epoch duration is 10 minutes.
```

Comment:
Any specific reason 10m was chosen? What is the trade-off for choosing a shorter or longer one?

Reply:
>  traceablity and circuit building overhead.

gotcha